hack/common.sh

```
@@ -135,3 +135,37 @@ function tenant::deploy_snapshotresources() {
+    if ! $KUBE_CMD rollout status "$RES_TYPE/$RES_NAME" -n "$NS" --timeout="$TIMEOUT"; then
+        echo "Error: Rollout failed"
+
+        # 1. Describe the failing resource
+        $KUBE_CMD describe "$RES_TYPE/$RES_NAME" -n "$NS"
+
+        # 2. Get Pods specifically for this app
+        # We assume the resource name matches part of the pod name.
+        $KUBE_CMD get pods -n "$NS" -o wide | grep "$RES_NAME"
+
+        # 3. Fetch logs from the first failing pod found
+        local POD_NAME=$($KUBE_CMD get pods -n "$NS" | grep "$RES_NAME" | awk '{print $1}' | head -n 1)
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)